aten/src/ATen/native/ComplexHelper.h

```diff
@@ -13,17 +13,50 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
   return res;
 }
 
-// expects as input a complex tensor and returns back a float tensor
-// containing the complex values in the last two dimensions
-inline Tensor view_complex_as_float(const Tensor& self) {
-  TORCH_INTERNAL_ASSERT(self.is_complex());
+// expects as input a complex tensor and returns back a tensor
+// with corresponding real dtype containing the complex values
+// in the last two dimensions
+Tensor view_as_real(const Tensor& self) {
+  TORCH_CHECK(self.is_complex(), "view_as_real is only supported for complex tensors");
   auto new_sizes = self.sizes().vec();
   // last dimension will always have two elements containing the real and imag vals
   new_sizes.emplace_back(2);
-  auto new_strides = computeStrideForComplex(self.strides());
+  auto strides = self.strides().vec();
+  auto new_strides = computeStrideForViewAsReal(self.strides());
   auto new_storage_offset = 2 * self.storage_offset();
   const auto float_type = c10::toValueType(self.scalar_type());
   return at::empty({0}, self.options().dtype(float_type)).set_(self.storage(), new_storage_offset, new_sizes, new_strides);
 }
 
+inline std::vector<int64_t> computeStrideForViewAsComplex(IntArrayRef oldstride) {
+  auto res = oldstride.vec();
+  int dim = res.size();
+
+  TORCH_INTERNAL_ASSERT(res[dim-1] == 1 || res[dim-1] == 0);
```

Comment:
In the case where `res[dim-1] == 0` you don't need the division below right?

Reply:
sorry I don't follow. `res[dim -1]` is being removed from the tensor everytime

Reply:
But if the stride is 0, your conversion to complex will fail right. Because only a single floating point number is in that dimension, not two as you expect.

Reply:
I don't think so because you could have tensors of some arbitrary shape just viewing one value and you should still be able to view that tensor as complex or real. for eg. `grad` is initialized in a similar way